File: csharp/Darts.cs
using System;

public static class Darts
{
    public static int Score(double x, double y) =>
        Hypot(x, y) switch 
        {
            > 10.0 => 0,
            > 5.0 => 1,
            > 1.0 => 5,
            _ => 10
        };

    private static double Hypot(double x, double y) => Math.Sqrt(Math.Pow(x, 2) + Math.Pow(y, 2));
}

File: haskell/Darts.hs
module Darts (score) where

score :: Float -> Float -> Int
score x y
  | z <= 1 = 10
  | z <= 5 = 5
  | z <= 10 = 1
  | otherwise = 0
  where
    z = sqrt (x ^ 2 + y ^ 2)
